buffer/src/test/java/io/netty/buffer/CompositeByteBufTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+import org.junit.Test;
+
+import static io.netty.buffer.Unpooled.*;
+import static org.junit.Assert.*;
+
+public class CompositeByteBufTest extends AdvancedLeakAwareByteBufTest {
+
+    @Test
+    public void testAddComponentWithLeakAwareByteBuf() {
+        ByteBuf buffer = newBuffer("hello world".length());
```

Comment:
are you sure this is correct ? It seems like this will only allocate an "empty" buffer with the length of "hello world".